spec/tunes/app_version_spec.rb

```diff
@@ -150,10 +171,308 @@
+        expect(trailer.descriptionXML).to match(/FoghornLeghorn/)
+        expect(trailer.preview_frame_time_code).to eq("00:00:12:34")
+        expect(trailer.video_url).to eq(nil)
+        expect(trailer.preview_image_url).to eq(nil)
+        expect(trailer.full_sized_preview_image_url).to eq(nil)
+        expect(trailer.device_type).to eq("ipad")
+        expect(trailer.language).to eq("English")
+      end
+
+      # TODO: can we detect trailer source change ?
+
+      it "remove the video trailer" do
+        count = ipad_trailers.count
+        expect(count).to eq(1)
+        version.upload_trailer!(nil, "English", 'ipad')
+        count_after = ipad_trailers.count
+        expect(count_after).to eq(count - 1)
+      end
+    end
+
+    describe "Reading and modifying the geojson file", :du do
+      before do
+        json = JSON.parse(du_read_upload_geojson_response_success)
+        allow(client.du_client).to receive(:upload_geojson).and_return(json)
+      end
+
+      it "default geojson data is nil when value field is missing" do
+        expect(version.raw_data["transitAppFile"]["value"]).to eq(nil)
+        expect(version.transit_app_file).to eq(nil)
+      end
+
+      it "modifies the geojson file data after update" do
+        version.upload_geojson!(du_fixture_file_path("upload_valid.geojson"))
+        expect(version.transit_app_file.name).to eq("upload_valid.geojson")
+        expect(version.transit_app_file.url).to eq(nil)
+        expect(version.transit_app_file.asset_token).to eq("Purple1/v4/45/50/9d/45509d39-6a5d-7f55-f919-0fbc7436be61/pr_source.geojson")
+      end
+
+      it "deletes the geojson" do
+        version.upload_geojson!(du_fixture_file_path("upload_valid.geojson"))
+        version.upload_geojson!(nil)
+        expect(version.raw_data["transitAppFile"]["value"]).to eq(nil)
+        expect(version.transit_app_file).to eq(nil)
+      end
+    end
+
+    describe "Upload screenshots", :screenshots do
+      before do
+        allow(Spaceship::UploadFile).to receive(:from_path) do |path|
+          du_uploadimage_correct_screenshot if path == "path_to_screenshot"
+        end
+      end
+      let(:screenshot_path) { "path_to_screenshot" }
+
+      describe "Parameter checks" do
+        it "prevents from using negative sort_order" do
+          expect do
+            version.upload_screenshot!(screenshot_path, -1, "English", 'iphone4')
+          end.to raise_error "sort_order must be positive"
+        end
+
+        it "prevents from using sort_order 0" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 0, "English", 'iphone4')
+          end.to raise_error "sort_order must be positive"
+        end
+
+        it "prevents from using too large sort_order" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 6, "English", 'iphone4')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.